pymemcache/serde.py

```diff
@@ -17,82 +17,128 @@
-        output = BytesIO()
-        pickler = pickle.Pickler(output, PICKLE_VERSION)
-        pickler.dump(value)
-        value = output.getvalue()
+    def from_python(self, key, value):
+        """
+        Serialize a python object.
 
-    return value, flags
+        :param str|unicode key: Key
+        :param str|unicode value: Value
+        :return tuple[str, str]: tuple(value, flags)
+        """
+        flags = 0
+        value_type = type(value)
 
+        # Check against exact types so that subclasses of native types will be
+        # restored as their native type
+        if value_type is bytes:
+            pass
 
-def python_memcache_deserializer(key, value, flags):
-    if flags == 0:
-        return value
+        elif value_type is six.text_type:
+            flags |= FLAG_TEXT
+            value = value.encode('utf8')
 
-    elif flags & FLAG_TEXT:
-        return value.decode('utf8')
+        elif value_type is int:
+            flags |= FLAG_INTEGER
+            value = "%d" % value
 
-    elif flags & FLAG_INTEGER:
-        return int(value)
+        elif six.PY2 and value_type is long_type:
+            flags |= FLAG_LONG
+            value = "%d" % value
 
-    elif flags & FLAG_LONG:
-        if six.PY3:
-            return int(value)
         else:
-            return long_type(value)
-
-    elif flags & FLAG_PICKLE:
-        try:
-            buf = BytesIO(value)
-            unpickler = pickle.Unpickler(buf)
-            return unpickler.load()
-        except Exception:
-            logging.info('Pickle error', exc_info=True)
-            return None
-
-    return value
+            flags |= FLAG_PICKLE
+
+            output = BytesIO()
+            pickler = pickle.Pickler(output, self.pickle_version)
+            pickler.dump(value)
+            value = output.getvalue()
+
+        return value, flags
+
+    def to_python(self, key, value, flags):
+        """
+        Deserialize a value into a python object.
+
+        :param str|unicode key: Key
+        :param str|unicode value: Value
+        :param int flags: Bitflag containing flags used to specify how to
+        deserialize this object.
+        :return object: Deserialized python object.
+        """
+        if flags == 0:
+            return value
+
+        elif flags & FLAG_TEXT:
+            return value.decode('utf8')
+
+        elif flags & FLAG_INTEGER:
+            return int(value)
+
+        elif flags & FLAG_LONG:
+            if six.PY3:
+                return int(value)
+            else:
+                return long_type(value)
+
+        elif flags & FLAG_PICKLE:
+            try:
+                buf = BytesIO(value)
+                unpickler = pickle.Unpickler(buf)
+                return unpickler.load()
+            except Exception as exc:
+                # This includes exc as a string for troubleshooting as well as providing
+                # a trace.
+                log.exception('Could not depickle value: %s')
```

Comment:
I changed this because using the root logger is nearly impossible to troubleshoot, and this should really be exc level.